frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx

```diff
@@ -70,13 +70,6 @@ export function PropertyValue({
     useEffect(() => {
         if (!value) {
             setInput('')
-        } else if (value !== input) {
```

Comment:
I genuinely cannot work out what value this code has - I've removed it and tried a number of scenarios and cannot see any adverse effects - it'd be amazing if @alexkim205 you might be able to share any context on why this might be here.

I've removed it because it causes this bug in production where suggest an option which is equivalent to a concatenated string of all the existing values.

![image](https://user-images.githubusercontent.com/85295485/127502332-4fef37c0-7064-431c-b733-264e2e4f2b46.png)

Reply:
Tested too and couldn't find any issues, I think it should be safe to remove. Tagging @samwinslow in case you have any context too

Reply:
It looks like it might be trying to do something if the values are JSON objects - but I can't think why the values would be JSON objects - perhaps I'm missing something?

Reply:
Looked at the code briefly and I think removing this JSON check might break functionality in use cases where we pass in a `value` property that represents a filter id. For example, `<PropertyValue/>` is still being used like this in [Path Tab](https://github.com/PostHog/posthog/blob/7667d128e815bf6f135a029c52c2298633f4dc4a/frontend/src/scenes/insights/InsightTabs/PathTab.tsx#L59-L59) where we're passing a numeric `filter.start_point` in as `value` which will always trigger this code path. This will set the input text as a numeric value on refresh/`value` change, instead of the name of the property associated with that numeric id.

I don't think we should remove this until we fully deprecate the other consumers of this component, or move this logic one up the component tree.

Reply:
Great catch! Let's bring back this code!